components/services/bat_ledger/bat_ledger_impl.cc

```diff
@@ -287,19 +287,19 @@ void BatLedgerImpl::OnTimer(uint32_t timer_id) {
 void BatLedgerImpl::GetAllBalanceReports(
     GetAllBalanceReportsCallback callback) {
   auto reports = ledger_->GetAllBalanceReports();
-  base::flat_map<std::string, std::string> out_reports;
+  base::flat_map<std::string, ledger::BalanceReportInfoPtr> out_reports;
```

Comment:
could we just do here something like:

```
void BatLedgerImpl::GetAllBalanceReports(GetAllBalanceReportsCallback callback) {
  auto reports = ledger_->GetAllBalanceReports();
  auto out_reports = mojo::MapToFlatMap(reports):
  std::move(callback).Run(std::move(out_reports));
}
```

Reply:
Problem is that `GetAllBalanceReportsCallback` expects the value type to be `BalanceReportInfoPtr`, but `MapToFlatMap` will give a `base::flatt_map<std::string, ledger::BalanceReportInfo>`

Reply:
`MapToFlatMap` doesn't want to work on this type, but you can work around it.